src/ngx_str.hpp

```diff
@@ -0,0 +1,19 @@
+#pragma once
```

Comment:
I wonder if we should call this header `ngx.hpp` instead and make a habit of including it as the first one in every cpp file. I remember, `ngx_*.h` headers were causing some errors if not included first.

Reply:
See [ccb30e1](https://github.com/nginxinc/nginx-otel/pull/34/commits/ccb30e11659828c2fb9efa0229cecee0cdcb4237).